### PR TITLE
Lite 18595 create product functionality

### PR DESCRIPTION
**Pull request by martinconstante**

### CODE REVIEW

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCCapabilitiesPpu implements CBCEntity {
+
+    @JsonProperty
+    private String schema;
+
+    @JsonProperty
+    private String dynamic;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java:16]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    @JsonProperty
+    private String predictive;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java:19]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    
+    @JsonProperty
+    private String future;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java:22]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesReservation.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCCapabilitiesReservation implements CBCEntity {
+
+    @JsonProperty
+    private String consumption;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesReservation.java:13]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesValidation.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCCapabilitiesValidation implements CBCEntity {
+
+    @JsonProperty
+    private String validation;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesValidation.java:13]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCCommitment.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCCommitment implements CBCEntity {
+
+    @JsonProperty
+    private String multiplier;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCommitment.java:13]: Integer

File: src/main/java/com/cloudblue/connect/api/models/CBCCommitment.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    @JsonProperty
+    private String count;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCCommitment.java:16]: Integer

File: src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import com.cloudblue.connect.api.models.common.CBCEntity;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCConfigurations implements CBCEntity {
+    @JsonProperty
+    private String suspendResumeSupported;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java:12]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    
+    @JsonProperty
+    private String requiresResellerInformation;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java:15]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCExtensions.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    private String id;
+    
+    @JsonProperty
+    private String icon;
+
+    @JsonProperty
+    private String name;
+
+    @JsonProperty
+    private String status;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCExtensions.java:21]: Should be of type Enum with all possible values.

File: src/main/java/com/cloudblue/connect/api/models/CBCItemUi.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCItemUi implements CBCEntity {
+
+    @JsonProperty
+    private String visibility;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCItemUi.java:13]: Boolean

File: src/main/java/com/cloudblue/connect/api/models/CBCUsageRule.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CBCUsageRule implements CBCEntity {
+
+    @JsonProperty
+    private String enabled;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/models/CBCUsageRule.java:15]: Boolean

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+    @Expression
+    private String mpn;
+
+    @Parameter
+    @Expression
+    private String description;
+
+    @Parameter
+    @Expression
+    private String type;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:32]: Should be Enum with possible values.

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+
+    @Parameter
+    @Expression
+    private String dynamic;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:36]: Boolean

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+
+    @Parameter
+    @Expression
+    private String period;
+
+    @Parameter
+    @Expression
+    private String commitmentCount;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:44]: Integer

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+
+    @Parameter
+    @Expression
+    private String unitId;
+
+    @Parameter
+    @Expression
+    private String precision;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:52]: Should be double or integer

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+
+    @Parameter
+    @Expression
+    private String uiVisibility;
```
Comment (rahulmondal) [inline src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:56]: Boolean?

**Review by rahulmondal — CHANGES_REQUESTED** [review]

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java (hunk 1 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java:16]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java (hunk 2 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java:19]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java (hunk 3 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesPpu.java:22]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesReservation.java (hunk 4 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesReservation.java:13]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesValidation.java (hunk 5 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCCapabilitiesValidation.java:13]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCCommitment.java (hunk 7 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCCommitment.java:16]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java (hunk 8 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java:12]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCExtensions.java (hunk 10 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCExtensions.java:21]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java (hunk 9 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCConfigurations.java:15]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCItemUi.java (hunk 11 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCItemUi.java:13]: Done

File: src/main/java/com/cloudblue/connect/api/models/CBCUsageRule.java (hunk 12 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/models/CBCUsageRule.java:15]: Done

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java (hunk 14 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:36]: Done

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java (hunk 15 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:44]: Done

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java (hunk 16 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:52]: Done

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java (hunk 17 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java:56]: yes. Done

File: src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java (hunk 13 above)
Comment (martinconstante) [reply src/main/java/com/cloudblue/connect/api/parameters/products/NewProductItemParameter.java]: Done

**Review by rahulmondal — APPROVED** [review]

**Pull request merged**